### PR TITLE
fix nodelifecyle controller not add NoExecute taint bug

### LINKED ISSUE #94183: nodelifecycle controller does not set NoExecute taint to NotReady node when kubelet has been stopped
**What happened**:
```shell
[root@liabio ~]# date
Sun Aug 23 22:28:31 HKT 2020
[root@liabio ~]# kubectl  get no liabio -oyaml
apiVersion: v1
kind: Node
metadata:
  annotations:
    kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
    node.alpha.kubernetes.io/ttl: "0"
    projectcalico.org/IPv4Address: 172.18.1.1/16
    projectcalico.org/IPv4IPIPTunnelAddr: 192.168.155.64
    volumes.kubernetes.io/controller-managed-attach-detach: "true"
  creationTimestamp: "2020-07-29T16:06:17Z"
  labels:
    beta.kubernetes.io/arch: amd64
    beta.kubernetes.io/os: linux
    kubernetes.io/arch: amd64
    kubernetes.io/hostname: liabio
    kubernetes.io/os: linux
    node-role.kubernetes.io/master: ""
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:kubeadm.alpha.kubernetes.io/cri-socket: {}
        f:labels:
          f:node-role.kubernetes.io/master: {}
    manager: kubeadm
    operation: Update
    time: "2020-07-29T16:06:20Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:projectcalico.org/IPv4Address: {}
          f:projectcalico.org/IPv4IPIPTunnelAddr: {}
      f:status:
        f:conditions:
          k:{"type":"NetworkUnavailable"}:
            .: {}
            f:lastHeartbeatTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
    manager: calico-node
    operation: Update
    time: "2020-08-02T16:57:10Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:volumes.kubernetes.io/controller-managed-attach-detach: {}
        f:labels:
          .: {}
          f:beta.kubernetes.io/arch: {}
          f:beta.kubernetes.io/os: {}
          f:kubernetes.io/arch: {}
          f:kubernetes.io/hostname: {}
          f:kubernetes.io/os: {}
      f:status:
        f:addresses:
          .: {}
          k:{"type":"Hostname"}:
            .: {}
            f:address: {}
            f:type: {}
          k:{"type":"InternalIP"}:
            .: {}
            f:address: {}
            f:type: {}
        f:allocatable:
          .: {}
          f:cpu: {}
          f:ephemeral-storage: {}
          f:hugepages-1Gi: {}
          f:hugepages-2Mi: {}
          f:memory: {}
          f:pods: {}
        f:capacity:
          .: {}
          f:cpu: {}
          f:ephemeral-storage: {}
          f:hugepages-1Gi: {}
          f:hugepages-2Mi: {}
          f:memory: {}
          f:pods: {}
        f:conditions:
          .: {}
          k:{"type":"DiskPressure"}:
            .: {}
            f:lastHeartbeatTime: {}
            f:type: {}
          k:{"type":"MemoryPressure"}:
            .: {}
            f:lastHeartbeatTime: {}
            f:type: {}
          k:{"type":"PIDPressure"}:
            .: {}
            f:lastHeartbeatTime: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastHeartbeatTime: {}
            f:type: {}
        f:daemonEndpoints:
          f:kubeletEndpoint:
            f:Port: {}
        f:images: {}
        f:nodeInfo:
          f:architecture: {}
          f:bootID: {}
          f:containerRuntimeVersion: {}
          f:kernelVersion: {}
          f:kubeProxyVersion: {}
          f:kubeletVersion: {}
          f:machineID: {}
          f:operatingSystem: {}
          f:osImage: {}
          f:systemUUID: {}
    manager: kubelet
    operation: Update
    time: "2020-08-23T14:08:25Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:node.alpha.kubernetes.io/ttl: {}
      f:spec:
        f:podCIDR: {}
        f:podCIDRs:
          .: {}
          v:"192.168.0.0/24": {}
        f:taints: {}
      f:status:
        f:conditions:
          k:{"type":"DiskPressure"}:
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
          k:{"type":"MemoryPressure"}:
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
          k:{"type":"PIDPressure"}:
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
          k:{"type":"Ready"}:
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
    manager: kube-controller-manager
    operation: Update
    time: "2020-08-23T14:11:07Z"
  name: liabio
  resourceVersion: "4969965"
  selfLink: /api/v1/nodes/liabio
  uid: 03b3a563-1905-4c11-95ea-3197024aee17
spec:
  podCIDR: 192.168.0.0/24
  podCIDRs:
  - 192.168.0.0/24
  taints:
  - effect: NoSchedule
    key: node.kubernetes.io/unreachable
    timeAdded: "2020-08-23T14:11:07Z"
status:
  addresses:
  - address: 10.0.9.52
    type: InternalIP
  - address: liabio
    type: Hostname
  allocatable:
    cpu: "2"
    ephemeral-storage: "38644306266"
    hugepages-1Gi: "0"
    hugepages-2Mi: "0"
    memory: 3745392Ki
    pods: "110"
  capacity:
    cpu: "2"
    ephemeral-storage: 41931756Ki
    hugepages-1Gi: "0"
    hugepages-2Mi: "0"
    memory: 3847792Ki
    pods: "110"
  conditions:
  - lastHeartbeatTime: "2020-08-02T16:57:10Z"
    lastTransitionTime: "2020-08-02T16:57:10Z"
    message: Calico is running on this node
    reason: CalicoIsUp
    status: "False"
    type: NetworkUnavailable
  - lastHeartbeatTime: "2020-08-23T14:08:25Z"
    lastTransitionTime: "2020-08-23T14:11:07Z"
    message: Kubelet stopped posting node status.
    reason: NodeStatusUnknown
    status: Unknown
    type: MemoryPressure
  - lastHeartbeatTime: "2020-08-23T14:08:25Z"
    lastTransitionTime: "2020-08-23T14:11:07Z"
    message: Kubelet stopped posting node status.
    reason: NodeStatusUnknown
    status: Unknown
    type: DiskPressure
  - lastHeartbeatTime: "2020-08-23T14:08:25Z"
    lastTransitionTime: "2020-08-23T14:11:07Z"
    message: Kubelet stopped posting node status.
    reason: NodeStatusUnknown
    status: Unknown
    type: PIDPressure
  - lastHeartbeatTime: "2020-08-23T14:08:25Z"
    lastTransitionTime: "2020-08-23T14:11:07Z"
    message: Kubelet stopped posting node status.
    reason: NodeStatusUnknown
    status: Unknown
    type: Ready
  daemonEndpoints:
    kubeletEndpoint:
      Port: 10250

```

Nodelifecycle controller does not set NoExecute taint to NotReady node when kubelet has been stopped

**What you expected to happen**:
- effect: NoExecute
    key: node.kubernetes.io/unreachable
Nodelifecycle controller sets NoExecute taint to NotReady node when kubelet has been stopped

**How to reproduce it (as minimally and precisely as possible)**:
systemctl stop kubelet

**Anything else we need to know?**:
I use default args to start kube-controller-manager.
```yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  labels:
    component: kube-controller-manager
    tier: control-plane
  name: kube-controller-manager
  namespace: kube-system
spec:
  containers:
  - command:
    - kube-controller-manager
    - --allocate-node-cidrs=true
    - --authentication-kubeconfig=/etc/kubernetes/controller-manager.conf
    - --authorization-kubeconfig=/etc/kubernetes/controller-manager.conf
    - --bind-address=127.0.0.1
    - --client-ca-file=/etc/kubernetes/pki/ca.crt
    - --cluster-cidr=192.168.0.0/16
    - --cluster-name=kubernetes
    - --cluster-signing-cert-file=/etc/kubernetes/pki/ca.crt
    - --cluster-signing-key-file=/etc/kubernetes/pki/ca.key
    - --controllers=*,bootstrapsigner,tokencleaner
    - --kubeconfig=/etc/kubernetes/controller-manager.conf
    - --leader-elect=true
    - --node-cidr-mask-size=24
    - --port=0
    - --requestheader-client-ca-file=/etc/kubernetes/pki/front-proxy-ca.crt
    - --root-ca-file=/etc/kubernetes/pki/ca.crt
    - --service-account-private-key-file=/etc/kubernetes/pki/sa.key
    - --service-cluster-ip-range=10.10.0.0/16
    - --use-service-account-credentials=true
    image: k8s.gcr.io/kube-controller-manager:v1.18.6
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 8
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10257
        scheme: HTTPS
      initialDelaySeconds: 15
      timeoutSeconds: 15
    name: kube-controller-manager
    resources:
      requests:
        cpu: 200m
    volumeMounts:
    - mountPath: /etc/ssl/certs
      name: ca-certs
      readOnly: true
    - mountPath: /etc/pki
      name: etc-pki
      readOnly: true
    - mountPath: /usr/libexec/kubernetes/kubelet-plugins/volume/exec
      name: flexvolume-dir
    - mountPath: /etc/kubernetes/pki
      name: k8s-certs
      readOnly: true
    - mountPath: /etc/kubernetes/controller-manager.conf
      name: kubeconfig
      readOnly: true
  hostNetwork: true
  priorityClassName: system-cluster-critical
  volumes:
  - hostPath:
      path: /etc/ssl/certs
      type: DirectoryOrCreate
    name: ca-certs
  - hostPath:
      path: /etc/pki
      type: DirectoryOrCreate
    name: etc-pki
  - hostPath:
      path: /usr/libexec/kubernetes/kubelet-plugins/volume/exec
      type: DirectoryOrCreate
    name: flexvolume-dir
  - hostPath:
      path: /etc/kubernetes/pki
      type: DirectoryOrCreate
    name: k8s-certs
  - hostPath:
      path: /etc/kubernetes/controller-manager.conf
      type: FileOrCreate
    name: kubeconfig
status: {}
```

I see log from kube-controller-manager:
https://github.com/kubernetes/kubernetes/blob/dff82dc0de47299ab66c83c626e08b245ab19037/pkg/controller/nodelifecycle/scheduler/rate_limited_queue.go#L239

**Environment**:
- Kubernetes version (use `kubectl version`):
[root@liabio ~]# kubectl  version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
[root@liabio ~]# cat /etc/os-release
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"

- Kernel (e.g. `uname -a`):
[root@liabio ~]# uname -a
Linux liabio 4.18.0-147.5.1.el8_1.x86_64 #1 SMP Wed Feb 5 02:00:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by howieyuen**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
this PR https://github.com/kubernetes/kubernetes/pull/89059 try to fix reconcile problem, so every 5s `monitorNodeHealth()`  run `processTaintBaseEviction()`, add nodes to `zoneNoExecuteTainter` cause these nodes' status is `Unknown` or `False`. 

However, every time we need add untainted nodes to `RateLimitedTimedQueue`, this PR try to delete it first in order to enter queue every time. Delete action use a additional func `SetRemve()` as below instead of `Remove()`:
```go
func (q *UniqueQueue) SetRemove(value string) {
	q.lock.Lock()
	defer q.lock.Unlock()
        // *************************************************
        // only delete set value, leave queue data behind
        // *************************************************
	if q.set.Has(value) {
		q.set.Delete(value)
	}
}

func (q *UniqueQueue) Remove(value string) bool {
	q.lock.Lock()
	defer q.lock.Unlock()

	if !q.set.Has(value) {
		return false
	}
	q.set.Delete(value)
	for i, val := range q.queue {
		if val.Value == value {
			heap.Remove(&q.queue, i)
			return true
		}
	}
	return true
}
```
When taintManager start working(`doNoExecuteTaintingPass()`) and its QPS defaults as `0.1`, so here is a scenario may case nodes will never get `NoExecute` taint except kube-controller-manager restart and reconstruct its queue data:

1. Suppose there are 3 nodes' status are unknown, and monitor these 3 node, add them in queue, the `UniqueQueue` inside `RateLimitedTimedQueue` looks like this:

| queue |  TimedValue{node0, id0} |  TimedValue{node1, id1} |  TimedValue{node2, id2} | 
| ------- | --------------------------- | -------------------------- | --------------------------- |
| set      |  node0                              |  node1                            |  node2                             | 

2. `doNoExecuteTaintingPass()` not finish the taint job, and `monitNodeHealth()` run in next period, and enqueue 3 nodes again with set data removed but queue data left, and the `UniqueQueue` inside `RateLimitedTimedQueue` looks like this:

| queue |  TimedValue{node0, id0}(dirty data) |  TimedValue{node1, id1}(dirty data) |  TimedValue{node2, id2}(dirty data)  | TimedValue{node0, id0} |  TimedValue{node1, id1} |  TimedValue{node2, id2} | 
| ---- | ---- | ---- | ---- | ---- |  ---- |  ---- |  
| set  | ~~node0~~ |  ~~node1~~ | ~~node2~~| node0 |  node1 | node2 |

3. `doNoExecuteTaintingPass()` continue to deal with these untainted nodes, this func fetch data from queue not set one by one, start with dirty data, suppose that before handling the duplicated data "node0", node0 return normal (same as node1 and node2), so `ActionFunc()` in `nc.zoneNoExecuteTainter[k].Try(fn ActionFunc)` returns true, and func `Try()` calls `q.queue.RemoveFromQueue(val.Value)`, but it cannot be removed because the set value is not existed. So queue's head cannot be removed normally, next running circle still get the dirty data, and the taint job go stuck forever
```go
func (q *UniqueQueue) RemoveFromQueue(value string) bool {
	q.lock.Lock()
	defer q.lock.Unlock()

	if !q.set.Has(value) {
		return false
	}
	for i, val := range q.queue {
		if val.Value == value {
			heap.Remove(&q.queue, i)
			return true
		}
	}
	return false
}
```

**Which issue(s) this PR fixes**:
Fix:  #94183 #96183

**Special notes for your reviewer**:
I write a helper func to print values inside `RateLimitedTimedQueue`, and unit test running log as below, and log display the dirty data inside queue field.

before set `SetRemove()` to `Remove()`
```
TestApplyNoExecuteTaintsToNodesEnqueueTwice
=== RUN   TestApplyNoExecuteTaintsToNodesEnqueueTwice
I1126 20:27:44.590489   60621 node_lifecycle_controller.go:380] Sending events to api server.
I1126 20:27:44.591383   60621 taint_manager.go:163] Sending events to api server.
I1126 20:27:44.591550   60621 node_lifecycle_controller.go:508] Controller will reconcile labels.
I1126 20:27:44.591803   60621 node_lifecycle_controller.go:1428] Initializing eviction metric for zone: region1::zone1
W1126 20:27:44.591895   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node1. Assuming now as a timestamp.
W1126 20:27:44.592092   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node2. Assuming now as a timestamp.
W1126 20:27:44.592187   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node0. Assuming now as a timestamp.
I1126 20:27:44.592273   60621 node_lifecycle_controller.go:1244] Controller detected that zone region1::zone1 is now in state Normal.
----------------------------------dirty data-----------------------------------
q.queue: node2,node0,node2,node0,
q.set: map[node0:{} node2:{}]
----------------------------------dirty data-----------------------------------
W1126 20:27:44.592663   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node3. Assuming now as a timestamp.
W1126 20:27:44.592753   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node4. Assuming now as a timestamp.
W1126 20:27:44.592817   60621 node_lifecycle_controller.go:1043] Missing timestamp for Node node5. Assuming now as a timestamp.
----------------------------------dirty data-----------------------------------
q.queue: node0,node0,node2,node5,node3,node0,
q.set: map[node0:{} node3:{} node5:{}]
map[node0:{} node3:{} node5:{}]
node0 true
map[node0:{} node3:{} node5:{}]
node2 false
----------------------------------dirty data-----------------------------------
    node_lifecycle_controller_test.go:291: Not found taint &Taint{Key:node.kubernetes.io/unreachable,Value:,Effect:NoExecute,TimeAdded:<nil>,} in [], which should be present in node3
    node_lifecycle_controller_test.go:299: Not found taint &Taint{Key:node.kubernetes.io/not-ready,Value:,Effect:NoExecute,TimeAdded:<nil>,} in [], which should be present in node5
--- FAIL: TestApplyNoExecuteTaintsToNodesEnqueueTwice (0.01s)
FAIL
FAIL	k8s.io/kubernetes/pkg/controller/nodelifecycle	0.030s
FAIL
```

After:
```
=== RUN   TestApplyNoExecuteTaintsToNodesEnqueueTwice
I1126 20:27:56.167537   36972 node_lifecycle_controller.go:380] Sending events to api server.
I1126 20:27:56.167790   36972 taint_manager.go:163] Sending events to api server.
I1126 20:27:56.167879   36972 node_lifecycle_controller.go:508] Controller will reconcile labels.
I1126 20:27:56.167988   36972 node_lifecycle_controller.go:1429] Initializing eviction metric for zone: region1::zone1
W1126 20:27:56.168036   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node0. Assuming now as a timestamp.
W1126 20:27:56.168097   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node1. Assuming now as a timestamp.
W1126 20:27:56.168139   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node2. Assuming now as a timestamp.
I1126 20:27:56.168184   36972 node_lifecycle_controller.go:1245] Controller detected that zone region1::zone1 is now in state Normal.
----------------------------------clean data-----------------------------------
q.queue: node0,node2,
q.set: map[node0:{} node2:{}]
----------------------------------clean data-----------------------------------
W1126 20:27:56.168407   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node3. Assuming now as a timestamp.
W1126 20:27:56.168447   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node4. Assuming now as a timestamp.
W1126 20:27:56.168488   36972 node_lifecycle_controller.go:1044] Missing timestamp for Node node5. Assuming now as a timestamp.
----------------------------------clean data-----------------------------------
q.queue: node2,node3,node5,
q.set: map[node2:{} node3:{} node5:{}]
node2 true
node3 true
node5 true
----------------------------------clean data-----------------------------------
--- PASS: TestApplyNoExecuteTaintsToNodesEnqueueTwice (0.00s)
PASS
ok  	k8s.io/kubernetes/pkg/controller/nodelifecycle	0.027s
```

**Does this PR introduce a user-facing change?**
```release-note
fixing a regression in 1.19+ where a failed node may not have the NoExecute taint set correctly
```

### CODE REVIEW

**Comment by howieyuen** [comment]
@k82cn 
@liggitt 
Please take a look, thanks for the reviewing 🍭

**Comment by howieyuen** [comment]
priority/important-soon

**Comment by howieyuen** [comment]
/assign @k82cn 
/assign @liggitt 
Please take a look, thanks for the reviewing 🍭

**Comment by howieyuen** [comment]
ping @k82cn @smarterclayton @liggitt @ingvagabund

**Comment by liggitt** [comment]
thanks for the fix and test, will route this to node reviewers

/unassign
/assign @derekwaynecarr 
/sig node

**Comment by howieyuen** [comment]
> thanks for the fix and test, will route this to node reviewers
> 
> /unassign
> /assign @derekwaynecarr
> /sig node

thanks for reply

**Comment by howieyuen** [comment]
ping @gmarek @bowei @k82cn
ping @smarterclayton @ingvagabund @aveshagarwal

**Comment by adampl** [comment]
This is a critical issue that already affects multiple major releases - 1.18.(something), 1.19 and 1.20, please do not delay!

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go [hunk 1]
```diff
@@ -2782,6 +2782,239 @@ func TestApplyNoExecuteTaints(t *testing.T) {
...
+
+	fakeNodeHandler := &testutil.FakeNodeHandler{
+		Existing: []*v1.Node{
+			// Unreachable Taint with effect 'NoExecute' should be applied to this node.
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "node0",
+					CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+					Labels: map[string]string{
+						v1.LabelZoneRegionStable:        "region1",
```
Comment (andrewsykim) [inline pkg/controller/nodelifecycle/node_lifecycle_controller_test.go:2798]: This is now an alias and the preferred constant is `LabelTopologyRegion`

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go [hunk 2]
```diff
@@ -2782,6 +2782,239 @@ func TestApplyNoExecuteTaints(t *testing.T) {
...
+						v1.LabelZoneFailureDomainStable: "zone1",
```
Comment (andrewsykim) [inline pkg/controller/nodelifecycle/node_lifecycle_controller_test.go:2799]: This is now an alias and the preferred constant is `LabelTopologyZone

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go [hunk 3]
```diff
@@ -2782,6 +2782,239 @@ func TestApplyNoExecuteTaints(t *testing.T) {
...
+						v1.LabelZoneRegion:              "region1",
```
Comment (andrewsykim) [inline pkg/controller/nodelifecycle/node_lifecycle_controller_test.go:2800]: This is now an alias and the preferred constant is `LabelFailureDomainBetaRegion`

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go [hunk 4]
```diff
@@ -2782,6 +2782,239 @@ func TestApplyNoExecuteTaints(t *testing.T) {
...
+						v1.LabelZoneFailureDomain:       "zone1",
```
Comment (andrewsykim) [inline pkg/controller/nodelifecycle/node_lifecycle_controller_test.go:2801]: This is now an alias and the preferred constant is `LabelFailureDomainBetaZone`

**Comment by andrewsykim** [comment]
Should this have release notes?

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go (hunk 1 above)
Comment (howieyuen) [reply pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: fixed, thanks for the review

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go (hunk 2 above)
Comment (howieyuen) [reply pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: fixed, thanks for the review

File: pkg/controller/nodelifecycle/node_lifecycle_controller_test.go (hunk 4 above)
Comment (howieyuen) [reply pkg/controller/nodelifecycle/node_lifecycle_controller_test.go]: fixed, thanks for the review

**Comment by howieyuen** [comment]
> Should this have release notes?

Yes, I think need one, because this memory leak bug did exist

**Comment by andrewsykim** [comment]
> Yes, I think need one, because this memory leak bug did exist

You can update the release notes by updating the section `Does this PR introduce a user-facing change?` in the PR description.

**Comment by adampl** [comment]
Come on guys, please complete the review, merge and start backporting ASAP.

**Comment by dims** [comment]
@adampl please have some patience, the right people are assigned.

**Comment by howieyuen** [comment]
> Come on guys, please complete the review, merge and start backporting ASAP.
thanks for promoting the PR, it get stuck for almost one month

**Comment by howieyuen** [comment]
> > Yes, I think need one, because this memory leak bug did exist
> 
> You can update the release notes by updating the section `Does this PR introduce a user-facing change?` in the PR description.

I may misunderstand the release not meaning, this PR doesn't contains a user-facing change, do we still need one ?

**Comment by andrewsykim** [comment]
> I may misunderstand the release note referring, this PR doesn't contains a user-facing change, do we still need one ?

This is fixing a bug where a failed node may not have the NoExecute taint set correctly. I think that bug is a user-facing change and the release notes should reflect the fix for it.

**Comment by howieyuen** [comment]
please take a look 
@k82cn @smarterclayton @liggitt @ingvagabund

**Comment by howieyuen** [comment]
@andrewsykim Can I get a lgtm ? 😄

**Comment by k82cn** [comment]
I'll help to review this PR ASAP :)

/assign

**Comment by kikisdeliveryservice** [comment]
> > I may misunderstand the release note referring, this PR doesn't contains a user-facing change, do we still need one ?
> 
> This is fixing a bug where a failed node may not have the NoExecute taint set correctly. I think that bug is a user-facing change and the release notes should reflect the fix for it.

hmm they updated the PR description with the release note text but the label is still unset.

/release-note

**Comment by kikisdeliveryservice** [comment]
@howieyuen Could you update/double check that your description is formatted correctly so the bot can pick up the label.

**Comment by howieyuen** [comment]
> @howieyuen Could you update/double check that your description is formatted correctly so the bot can pick up the label.

ok, I am checking it now

**Comment by derekwaynecarr** [comment]
thank you for the detail.

i have to admit the issue upon review was hard to follow without it, so the test and detail are much appreciated!

/approve
/lgtm

**Comment by LuckySB** [comment]
1.18.12 has this issue too, can you cherry-pick to 1.18 ?

**Comment by pacoxu** [comment]
Not sure if there's a reason that  cherry-picks prs are  not merged  to 1.18-1.20.
/cc
